internal/jptrace/aggregator.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package jptrace
+
+import (
+	"iter"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+// AggregateTraces aggregates a sequence of trace batches into individual traces.
+//
+// The `tracesSeq` input must adhere to the following requirements:
+//   - Each `Traces` object in the sequence must not be empty.
+//   - Each `Traces` object must contain at least one span
+//     (i.e., there must be at least one resource span with one scope span and one span).
+func AggregateTraces(tracesSeq iter.Seq2[[]ptrace.Traces, error]) iter.Seq2[ptrace.Traces, error] {
+	return func(yield func(trace ptrace.Traces, err error) bool) {
+		currentTrace := ptrace.NewTraces()
+		currentTraceID := pcommon.NewTraceIDEmpty()
+
+		tracesSeq(func(traces []ptrace.Traces, err error) bool {
+			if err != nil {
+				yield(ptrace.NewTraces(), err)
```

Comment:
@yurishkuro should we do an early return here or do we want to keep processing?

Reply:
i think we can return - the storage is not supposed to continue sending results if it returns an error. We can also specify that in the API contract.

Reply:
sounds good - looks like its already part of the reader api contract

```
If an error is encountered, the iterator returns the error and stops.
```